kubectl-plugin/pkg/cmd/submit/submit.go

```diff
@@ -0,0 +1,537 @@
+package submit
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/kubectl/pkg/cmd/portforward"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/google/shlex"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	"github.com/spf13/cobra"
+
+	rayv1api "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+)
+
+const dashboardAddr = "http://localhost:8265"
+
+type SubmitJobOptions struct {
+	ioStreams        *genericiooptions.IOStreams
+	configFlags      *genericclioptions.ConfigFlags
+	RayJob           *unstructured.Unstructured
+	submissionID     string
+	entryPoint       string
+	cluster          string
+	fileName         string
+	workingDir       string
+	runtimeEnv       string
+	address          string
+	entryPointCPU    float32
+	entryPointGPU    float32
+	entryPointMemory int
+	noWait           bool
+}
+
+type RayJob struct {
+	*rayv1api.RayJob
+}
+
+var (
+	submitJobLong = templates.LongDesc(`
+		Submit ray job to ray cluster as one would using ray CLI e.g. 'ray job submit ENTRYPOINT'. Command supports all options that 'ray job submit' supports.
+
+		If a RayJob CR is specified, command will apply RayJob CR and also submit the ray job.
+		RayJob CR is required when specifying working-dir within RayJob CR yaml or using --working-dir flag.
+
+		If ray address is set using --address or 'RAY_ADDRESS' env var, it will assume that there is already a way to connect to Ray cluster
+	`)
+
+	submitJobExample = templates.Examples(`
+		# Submit ray job as one would when using ray CLI
+		kubectl ray submit -- python my_script.py
```

Comment:
kubectl ray job submit

Reply:
Does 
```
kubectl ray submit -- python my_script.py
```

Actually work without passing in the RayJob? I think the examples should only include commands that will work

Reply:
Also do we want `kubectl ray submit job` or `kubectl ray job submit`? I was leaning towards `ray submit job` for consistency with kubectl as well as organization/structure